### PR TITLE
Feature/7510 error handling

### LINKED ISSUE #7510: WooCommerce Account Creation
### Description 
Create a new screen that given an email and password creates a new WordPress account

### Tasks

- [x] Add new "Get Started" button to prologue under feature flag that opens new Sign Up screen:  #7534 
- [x] Create new screen with given UI: #7534 
- [x] Add API requests for happy path: [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2544) and #7565
- [x] Error handling: #7584
- [x] Tracking #7673
- [x] Trigger store creation flow after successful signup  #7662 
- [x] Handle "Log in" CTA navigation to log in with WP.com credentials #7662 

### Designs 

Screenshot 2022-10-11 at 15 05 29

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Do not merge until #7565 is merged. 

Closes: #7510 

### Description
This adds needed changes to deal with all the possible errors from account creation. 

### Testing instructions
- Click on "Get Started" 
- Check button to proceed with account creation is disabled until email and password are not empty
- Type "dafsgdgs@gmail" as email and check -> Invalid email error is displayed
- Type "123" as password and check -> Password too short error
- Type "1234567" as password -> Check password not secure enough error
- Type "Jorgemucientes@gmail.com" as email and check -> email already exist
- Finally type a valid email and password and check account is created successfully. 

### Images/gif

https://user-images.githubusercontent.com/2663464/197164371-484df3ba-66d1-435c-ae77-a66f2b5d7874.mp4

### CODE REVIEW

**Comment by hafizrahman** [comment]
> Type "[Jorgemucientes@gmail.com](mailto:Jorgemucientes@gmail.com)" as email and check -> email already exist

Just wanted to note that in WordPress.com, it has what I think is a more actionable error message. We might want to consider something similar in the future (especially since even in our current design we also already have similar design pattern of label text with tappable link for ToS):

Screen Shot 2022-10-21 at 19 57 40

**Review by hafizrahman — CHANGES_REQUESTED** [review]
In my test and around 0:16 in the video, the error message for the password field is also shown and replicated under the email address field. I think it makes more sense for it to only appear on the specific input field. Wdyt?

**Comment by hafizrahman** [comment]
❓ Is it indeed planned to have all validation done after pressing the "Get started" button? I can understand the existing email address checking will need network call, but curious if some checks can be done in real time instead (e.g: email/password length, password strength). Might be a bit better user experience, although definitely not a blocker.

**Comment by JorgeMucientes** [comment]
Thanks for the review @hafizrahman 🙏🏼 

> In my test and around 0:16 in the video, the error message for the password field is also shown and replicated under the email address field. I think it makes more sense for it to only appear on the specific input field. Wdyt?

Dammit, I don't know how I missed that bug, specially while I was doing the screen recording 😅 . Good catch 👍🏼. Fixed it! 

> Is it indeed planned to have all validation done after pressing the "Get started" button? ...

Good point. Yes! in fact, most of the validation are handled locally inside `validateCredentials()` from `SignUpRepository`. So, no API request is fired if those validations are not satisfied.  The fact that validations are triggered after clicking "Get started" is to make sure the user has finished entering the credentials. We could validate right after the Text field losses focus, but I don't see major gains from that. Wdyt?

**Comment by JorgeMucientes** [comment]
> Just wanted to note that in WordPress.com, it has what I think is a more actionable error message. We might want to consider something similar in the future (especially since even in our current design we also already have similar design pattern of label text with tappable link for ToS):

Yes, that would be a more actionable error, but kind of redundant as we are already going to offer  a direct CTA to "Log in" inside the screen (not implemented yet tough, waiting for final designs in Figma: HyVloP5FipZzyPVenH2euI-fi-1388%3A60209 )

File: WooCommerce/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -323,6 +323,11 @@
     <string name="signup_password_hint">Choose a password</string>
     <string name="signup_terms_of_service">By continuing, you agree to our &lt;u&gt;Terms of Service.&lt;/u&gt;</string>
     <string name="signup_creating_account_loading_message">Creating new account</string>
+    <string name="signup_email_exist_input">An account with this email already exists.</string>
+    <string name="signup_email_invalid_input">The email is invalid.</string>
```
Comment (hafizrahman) [inline WooCommerce/src/main/res/values/strings.xml:327]: Suggestion for a more actionable error message:

```suggestion
    <string name="signup_email_invalid_input">Please enter an invalid email address.</string>
```

**Review by hafizrahman — APPROVED** [review]

File: WooCommerce/src/main/res/values/strings.xml (hunk 1 above)
Comment (hafizrahman) [inline WooCommerce/src/main/res/values/strings.xml]: Suggestion for a bit more actionable error message.
```suggestion
    <string name="signup_email_invalid_input">Please enter a valid email address.</string>
```

**Comment by hafizrahman** [comment]
This now LGTM, I added a small error message improvement. Have a look and feel free to merge PR afterward.

>  The fact that validations are triggered after clicking "Get started" is to make sure the user has finished entering the credentials. We could validate right after the Text field losses focus, but I don't see major gains from that. Wdyt?

Yeah, I think there's trade-offs for either real-time validation (user can be sure they only submit correct values, but can also be slow on slower connection), or validation after submission (less networking-intensive, but if user makes several mistakes or unsure what their mistakes are, it's extra steps for them to submit over and over). But in any case there's no major gains for now, let's just go with what's already there :)

**Pull request merged**